src/imagery/i.landsat/i.landsat.download/i.landsat.download.py

```diff
@@ -206,108 +492,74 @@ def main():
-            dataset=options["dataset"],
-            bbox=bb,
-            start_date=start_date,
-            end_date=end_date,
-            max_cloud_cover=options["clouds"],
-            max_results=options["limit"],
-        )
-
-        if options["tier"]:
-            scenes = list(filter(lambda s: options["tier"] in s["display_id"], scenes))
-
-        # Output number of scenes found
-        gs.message(_("{} scenes found.".format(len(scenes))))
-
-        sort_vars = options["sort"].split(",")
-
-        reverse = False
-        if options["order"] == "desc":
-            reverse = True
-
-        # Sort scenes
-        sorted_scenes = sorted(
-            scenes, key=lambda i: (i[sort_vars[0]], i[sort_vars[1]]), reverse=reverse
-        )
-
-        landsat_api.logout()
-
-        if flags["l"]:
-
-            # Output sorted list of scenes found
-            # print('id', 'display_id', 'acquisition_date', 'cloud_cover')
-            for scene in sorted_scenes:
-                print(
-                    scene["entity_id"],
-                    scene["display_id"],
-                    scene["acquisition_date"].strftime("%Y-%m-%d"),
-                    scene["cloud_cover"],
-                )
-
-            gs.message(
-                _(
-                    "To download all scenes found, re-run the previous "
-                    "command without -l flag. Note that if no output "
-                    "option is provided, files will be downloaded in /tmp"
-                )
+        # Should use other provider other than USGS,
+        # as there was a bug in USGS API, fixed here
+        # https://github.com/CS-SI/eodag/issues/1252
+        # Currently availble in EODAG 3.0.0b3 (not yet released)
+        downloader.set_scenes_ids(options["id"].split(","))
+        # TODO: Remove when deserialize_and_register bug is fix
+        if options["datasource"] != "usgs":
+            downloader.download(outdir)
+        return 0
+
+    # Translate sort option to the i.eodag sort options
+    eodag_sort = ""
+    for sort_var in options["sort"].split(","):
+        if sort_var == "cloud_cover":
+            eodag_sort += "cloudcover,"
+        if sort_var == "acquisition_date":
+            eodag_sort += "ingestiondate,"
+
+    # Searching is necessary before listing or downloading
+    downloader.search(
+        dataset=options["dataset"],
+        start=start_date,
+        end=end_date,
+        clouds=options["clouds"] if options["clouds"] else None,
+        vector_map=options["map"] if options["map"] else None,
+        limit=options["limit"],
+        eodag_sort=eodag_sort,
+        eodag_order=options["order"],
+    )
+
+    if flags["l"]:
+        downloader.list()
+        gs.message(
+            _(
+                "To download all scenes found, re-run the previous "
+                "command without -l flag. Note that if no output "
+                "option is provided, files will be downloaded in /tmp"
```

Comment:
As Markus mentioned in another comment downloading to `/tmp` by default is at least unexpected.
Usually rather the current working directory is used as a default.
That said, I suggest to to remove the message. It does not provide any specific information that is not already in the manual...